miden-tx/src/testing/mod.rs

```diff
@@ -0,0 +1,260 @@
+use alloc::{string::ToString, vec::Vec};
+
+use miden_lib::transaction::{TransactionEvent, TransactionKernel};
+use miden_objects::{
+    accounts::{
+        account_id::testing::ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_OFF_CHAIN,
+        delta::AccountVaultDelta, Account, AccountDelta, AccountStub,
+    },
+    notes::Note,
+    testing::{
+        account::{mock_account, mock_new_account, MockAccountType},
+        account_code::mock_account_code,
+        build_dummy_tx_program,
+        notes::{mock_notes, AssetPreservationStatus},
+        storage::{mock_fungible_faucet, mock_non_fungible_faucet},
+    },
+    transaction::{
+        ChainMmr, ExecutedTransaction, InputNote, InputNotes, OutputNote, OutputNotes,
+        TransactionArgs, TransactionInputs, TransactionOutputs,
+    },
+    BlockHeader, Digest, FieldElement,
+};
+use vm_processor::{
+    crypto::NodeIndex, AdviceExtractor, AdviceInjector, AdviceInputs, AdviceProvider, AdviceSource,
+    ContextId, ExecutionError, Felt, Host, HostResponse, MemAdviceProvider, ProcessState, Word,
+};
+
+mod account_procs;
+pub mod chain_data;
+pub mod data_store;
+pub mod utils;
+
+use self::{account_procs::AccountProcedureIndexMap, chain_data::mock_chain_data};
+
+// MOCK HOST
+// ================================================================================================
+
+/// This is very similar to the TransactionHost in miden-tx. The differences include:
+/// - We do not track account delta here.
+/// - There is special handling of EMPTY_DIGEST in account procedure index map.
+/// - This host uses `MemAdviceProvider` which is instantiated from the passed in advice inputs.
+pub struct MockHost {
+    adv_provider: MemAdviceProvider,
+    acct_procedure_index_map: AccountProcedureIndexMap,
+}
```

Comment:
How is this different from the `MockHost` in `miden-tx/src/testing/mock_host.rs`.

Reply:
I had actually removed it but did not save the file and when pushing, I only pushed the new file. Removed.